routing/router.go

```diff
@@ -2163,15 +2164,29 @@ func (r *ChannelRouter) SendToRoute(htlcHash lntypes.Hash, rt *route.Route) (
 	// mark the payment failed with the control tower immediately. Process
 	// the error to check if it maps into a terminal error code, if not use
 	// a generic NO_ROUTE error.
-	if err := sh.handleSendError(attempt, shardError); err != nil {
-		return nil, err
-	}
+	var failureReason *channeldb.FailureReason
+	err = sh.handleSendError(attempt, shardError)
```

Comment:
I think the method should maybe just return `*channeldb.FailureReason` directly so it's clear that it can't return a "real" generic error.